ionic_user_interface/src/views/ViewRoute.vue

```diff
@@ -0,0 +1,548 @@
+                    .create({
+                      header: 'Your report has been successfully sent',
+                      position: 'bottom',
+                      color: 'success',
+                      duration: 3000,
+                      buttons: [
+                        {
+                          text: 'Close',
+                          role: 'cancel',
+                        },
+                      ],
+                    })
+                    .then((toast) => {
+                      toast.present();
+                    });
+                })
+                .catch((error) => {
+                  console.log(error);
+                  toastController
+                    .create({
+                      header: 'Failed to report user, please try again',
+                      position: 'bottom',
+                      color: 'danger',
+                      duration: 3000,
+                      buttons: [
+                        {
+                          text: 'Close',
+                          role: 'cancel',
+                        },
+                      ],
+                    })
+                    .then((toast) => {
+                      toast.present();
+                    });
+                });
+            },
+          },
+        ],
+      });
+      return alert.present();
+    }, 1000);
+
+    const reportRouteHandler = throttle(async (routeUsername: string, createdAt: string) => {
+      const alert = await alertController.create({
+        cssClass: 'wide',
+        header: `Report this route?`,
+        message: 'Are you sure?',
+        buttons: [
+          {
+            text: 'Cancel',
+            role: 'cancel',
+            cssClass: 'secondary',
+          },
+          {
+            text: 'Yes',
+            handler: () => {
+              axios
+                .post(
+                  process.env.VUE_APP_ROUTE_ENDPOINT_URL + '/route/details/report',
+                  {
+                    username: routeUsername,
+                    createdAt,
+                  },
+                  {
+                    headers: {
+                      Authorization: `Bearer ${getAccessToken().value}`,
+                    },
+                  },
+                )
+                .then((response) => {
+                  console.log(response);
+                  toastController
+                    .create({
+                      header: 'Your report has been successfully sent',
+                      position: 'bottom',
+                      color: 'success',
+                      duration: 3000,
+                      buttons: [
+                        {
+                          text: 'Close',
+                          role: 'cancel',
+                        },
+                      ],
+                    })
+                    .then((toast) => {
+                      toast.present();
+                    });
+                })
+                .catch((error) => {
```

Comment:
On reporting multiple times,

HTTP 400
```
{
    "Message": "Duplicate route report"
}
```

Reply:
I already disabled the button if `hasReport == true`